Early progression finalisation

LordIdra opened the pull request:
# WTF is this
Giant landmark PR intended to get Pylon into a playable state for the first time. This PR refines a LOT of stuff (mostly recipes and missing items/intermediates), with the summary being:
- Cleaned up some lore
- Added a bunch of new intermediates
- Added recipes to anything that didn't have recipes
- Changed many existing recipes to integrate better with everything else
- Lots of minor tweaks to eg item materials, guide categories, item durability, etc
- Added bill of materials to all multiblocks
- Overhauled researches

There are 100% going to be many bugs which have slipped through the cracks, though I've patched everything I've found and tested all my additions (except for testing all recipes obviously). We'll find and fix them with playtesting.

# Changes
- Crushed ore variants now need the hammer rather than grindstone
- Ore dusts can now be made from raw ores in the grindstone
- Removed silver, zinc, lead, brass (they're unused in early progression to avoid resource bloat - we can add them or other metals back later if we need them)
- Added raw variant for tin and added corresponding smelting recipe
- Added obsidian chip
- Added obscyra, a magical fluid made by mixing obsidian chips and blaze powder
- Added igneous composite, a material made with obscyra, covalent binder, and nether brick. This is the first material you can make which carries hot fluids
- Added rock dust recipes for granite, andesite, diorite
- Changed pit kiln to output 2 bronze for 2 copper + 1 tin
- Renamed carbon dust to carbon
- Added tin sheets, bronze sheets, steel sheets
- Added block variants of the new metals
- Yeeted ferroduralum (and changed anything that used it to use bronze, eg bronze tools)
- Added tin hammer
- Halved durability on hammers
- Changed a bunch of textures
- Removed compressed wood / compressed obsidian
- Updated portable items recipes
- Updated sprinkler recipe
- Updated lumber axe recipe to require magic altar
- Increase lumber axe durability (but made recipe harder)
- Slight buff to flour and dough recipe
- Changed covalent binder recipe
- Replaced redstone with tin dust in health talisman 1 recipe
- Yeeted plaster
- Changed bandage, splint, disinfectant, medkit recipes to make more sense
- Added some more lore to magic items - this is in no way final and can (and likely will) be changed later but thought it would be cool to start adding a little bit of magic lore
- Small wording improvements in a lot of lore lines
- Removed wither proof obsidian and made igneous composite witherproof instead
- Added pipes, portable tanks, and tank casings for wood, copper, tin, iron, bronze, igneous composite, steel
- Added nuggets and blocks for tin, bronze, steel, nickel, cobalt
- Standardised multiblock component lore to eg 'Smeltery multiblock component'
- Added recipes for fluid tank and fluid tank casing
- Added 'Fluid pipes & tanks' category and moves pipes/tanks to it
- Added 'Reflector Fluid' made with quartz dust and water for manufacturing solar lenses
- Added new intermediates: Rotor, hydraulic motor, backflow valve, analogue display, filter mesh, nozzle, abyssal catalyst, axle, sawblade, weighted shaft, copper drill bit, bronze drill bit
- Adjusted water pump, recipe
- Added recipes for fluid valve, fluid filter, fluid meter, water/lava placer/drainer, fluid voiders, 
- Removed dimensional barrel for now in favour of more complete storage solutions later down the line
- Changed (super)(fiery) explosive target recipes
- Changed elevator recipes (to use shimmer dust and fewer end pearls)
- Fixed elevators to be in the building crafting book category
- Made fireproof rune recipe cheaper (since it's not really worth it once you have netherite) and use magic altar
- Standardised recipe shapes to lowercase
- Adjusted ice arrow and slime ball recipes to both take and produce 8 arrows
- Added recipes for all hydraulic machines
- Increased coal purification tower height
- Added recipes for core miners
- Removed magic category and moved fireproof rune to tools
- Added gypsum and gypsum dust to gate smeltery, only obtainable using hydraulic core miner
- Added refractory mix, made with gypsum, sand, gravel, clay, and obscyra
- Added brick mold
- Added unfired refractory brick, made by holding right click with a brick mold in hand while looking at refractory mix
- Changed refractory brick to refractory bricks and added refractory brick, made by firing unfired refractory brick in pit kiln
- Made hammer recipes able to output multiple items
- Updated smeltery component recipes to make use of new intermediates
- Added bill of materials to all multiblocks
- Changed some fluid materials so all the metals are now different colors
- Added a cooldown to research packs. This is to incentivize going for higher tier research packs otherwise there might not be much point
- Overhauled a lot of researches. A lot of them are now more expensive to account for the new research packs. Notably, there are a set of items that are unlocked when you start, including grindstone, hammers, manual core drill, tin items, vanilla metal sheets/dusts, etc - just to give you some pointers when you start. Might have amped the cost too high but we'll find out later.
- Probably countless other things I forgot to write down

OhmV-IR commented on src/main/java/io/github/pylonmc/pylon/base/BaseResearches.java:
```diff
@@ -12,91 +12,100 @@ public class BaseResearches {
+                BaseKeys.HYDRAULIC_MOTOR
+        ).register();
+
+        new Research(
+                baseKey("scientific_revolution_1"),
+                Material.RED_BANNER,
+                10L,
+                BaseKeys.RESEARCH_PACK_1
         ).register();
```
Isn't it theoretically possible to softlock yourself then because if you run out of points from the loupe or whatever before you unlock this

Seggan replied:
It's ridiculously easy to get points from the loupe in the earlygame

LordIdra replied:
Yeah, it's only 10 points so I don't think this is realistically a concern

OhmV-IR commented on src/main/java/io/github/pylonmc/pylon/base/BaseResearches.java:
```diff
@@ -108,89 +117,309 @@ public void initialize() {
+                BaseKeys.IMMOBILIZER
+        ).register();
+
+        new Research(
+                baseKey("fluid_voiding"),
+                Material.BLACK_TERRACOTTA,
+                30L,
+                BaseKeys.FLUID_VOIDER_1,
+                BaseKeys.FLUID_VOIDER_2,
+                BaseKeys.FLUID_VOIDER_3
+        ).register();
```
Maybe split this one up into 2 so that the voider 1 can be cheaper

OhmV-IR reviewed (commented):
Not sure how necessary the BOM of multi blocks in the lore will be after PyCalc becomes a thing. Just cw comments in this review, but gj on 4k line pr

Seggan reviewed (changes requested):
- Core chunks don't have recipes in the guide
- No melting recipes for nuggets and blocks
- No guide recipe for unfired refractory brick

LordIdra commented on src/main/java/io/github/pylonmc/pylon/base/BaseFluids.java:
Don't think it's necessary since we use PylonFluid instead of the fluid keys everywhere.

LordIdra commented on src/main/java/io/github/pylonmc/pylon/base/content/tools/Moldable.java:
```diff
@@ -0,0 +1,10 @@
+public interface Moldable {
+    void doMoldingClick();
+    boolean isMoldingFinished();
+    ItemStack moldingResult();
```
we use both interchangeably, don't think it matters anyway

LordIdra commented on src/main/java/io/github/pylonmc/pylon/base/BaseItems.java:
Not a problem, it's a really tiny thing that's probably not even worth the changelog line anyway

LordIdra commented on src/main/java/io/github/pylonmc/pylon/base/BaseFluids.java:
```diff
@@ -50,57 +55,42 @@ private BaseFluids() {
 
     public static final PylonFluid GOLD = new PylonFluid(
             baseKey("gold"),
-            Material.GOLD_BLOCK
```
Consistency with all the other textures and also the texture is diverse enough that it looks really weird when stretched (as it is in fluid tanks)

LordIdra commented on src/main/java/io/github/pylonmc/pylon/base/content/machines/smelting/PitKiln.java:
My bad, misremembered the earlier convo about podzol

LordIdra commented:
> * Core chunks don't have recipes in the guide
Added fake drilling recipe to fix this

> * No melting recipes for nuggets and blocks
Done

> * No guide recipe for unfired refractory brick
Added fake molding recipe to fix this

Seggan reviewed (changes requested):
Just 2 things: `Moldable` should probably be a recipe instead, and the fake recipes should use `DisplayRecipeType` instead of custom recipe type

balugaq commented on src/main/java/io/github/pylonmc/pylon/base/BaseItems.java:
```diff
@@ -188,330 +355,389 @@ private BaseItems() {
+        PylonItem.register(PylonItem.class, TIN_INGOT);
+        BasePages.RESOURCES.addItem(BaseKeys.TIN_INGOT);
+
+        float cookingXp = Settings.get(BaseKeys.MONSTER_JERKY).getOrThrow("cooking.xp", Double.class).floatValue();
```
`BaseKeys.MONSTER_JERKY` ×
`BaseKeys.TIN_INGOT` √

balugaq commented:
Good job!

balugaq commented on src/main/resources/lang/en.yml:
```diff
@@ -833,35 +1057,55 @@ item:
+      <diamond> 1x <yellow>Chest</yellow>
+      <diamond> 1x <yellow><lang:pylon.pylonbase.item.hydraulic_core_drill_input_hatch.name></yellow>
+      <diamond> 1x <yellow><lang:pylon.pylonbase.item.hydraulic_core_drill_output_hatch.name></yellow>
+      <diamond> 2x <yellow><lang:pylon.pylonbase.item.copper_fluid_tank_casing.name></yellow>
```
Not `copper_fluid_tank_casing` but `fluid_tank_casing_copper`
